drivers/sensor/tmp108/tmp108.c

```diff
@@ -0,0 +1,350 @@
+		} else {
+			mode = TI_TMP108_FREQ_16_HZ;
+		}
+
+		if (tmp108_write_config(dev,
+					TI_TMP108_FREQ_MASK,
+					mode) < 0) {
+			return -EIO;
+		}
+		break;
+
+	case SENSOR_ATTR_SHUTDOWN_MODE:
+		if (tmp108_write_config(dev,
+					TI_TMP108_MODE_MASK,
+					TI_TMP108_MODE_SHUTDOWN) < 0) {
+			return -EIO;
+		}
+		drv_data->one_shot_mode = false;
+		break;
+
+	case SENSOR_ATTR_CONTINOUS_CONVERSION_MODE:
+		if (tmp108_write_config(dev,
+					TI_TMP108_MODE_MASK,
+					TI_TMP108_MODE_CONTINOUS) < 0) {
+			return -EIO;
+		}
+		drv_data->one_shot_mode = false;
+		break;
+
+	case SENSOR_ATTR_ONE_SHOT_MODE:
+		if (tmp108_write_config(dev,
+					TI_TMP108_MODE_MASK,
+					TI_TMP108_MODE_ONE_SHOT) < 0) {
+			return -EIO;
+		}
+		drv_data->one_shot_mode = true;
+		break;
+
+	case SENSOR_ATTR_ALERT_POLARITY:
+		if (val->val1 == 1) {
+			mode = TI_TMP108_CONF_POL_HIGH;
+		} else {
+			mode = TI_TMP108_CONF_POL_LOW;
+		}
+
+		if (tmp108_write_config(dev,
+					TI_TMP108_CONF_POL_MASK,
+					mode) < 0) {
+			return -EIO;
+		}
+		break;
+
+	default:
+		return -ENOTSUP;
+	}
+
+	return 0;
+}
+
+static const struct sensor_driver_api tmp108_driver_api = {
+	.attr_set = tmp108_attr_set,
+	.attr_get = tmp108_attr_get,
+	.sample_fetch = tmp108_sample_fetch,
+	.channel_get = tmp108_channel_get,
+	.trigger_set = tmp_108_trigger_set,
+};
+
+#ifdef  CONFIG_TMP108_ALERT_INTERRUPTS
+static int setup_interrupts(const struct device *dev)
+{
+	struct tmp108_data *drv_data = dev->data;
+
+	drv_data->gpio = device_get_binding(DT_INST_GPIO_LABEL(0, alert_gpios));
+
+	if (!drv_data->gpio) {
+		LOG_ERR("tmp108: gpio controller %s not found", DT_INST_GPIO_LABEL(0, alert_gpios));
+		return -EINVAL;
+	}
+
+	gpio_pin_configure(drv_data->gpio, DT_INST_GPIO_PIN(0, alert_gpios), GPIO_INPUT);
+
+	gpio_init_callback(&drv_data->temp_alert_gpio_cb,
+			   tmp108_trigger_handle_alert,
+			   BIT(DT_INST_GPIO_PIN(0, alert_gpios)));
+
+	gpio_add_callback(drv_data->gpio,
+			  &drv_data->temp_alert_gpio_cb);
+
+	gpio_pin_interrupt_configure(drv_data->gpio,
+				     DT_INST_GPIO_PIN(0, alert_gpios),
+				     GPIO_INT_EDGE_BOTH);
+	return 0;
+}
+#endif
+
+static int tmp108_init(const struct device *dev)
+{
+	struct tmp108_data *drv_data = dev->data;
+
+	/* Bind to the I2C bus that the sensor is connected */
+	drv_data->i2c_dev = device_get_binding(DT_INST_BUS_LABEL(0));
+
+	if (!drv_data->i2c_dev) {
+		LOG_ERR("Cannot bind to %s device!", DT_INST_BUS_LABEL(0));
+		return -EINVAL;
+	}
+
+	drv_data->scheduled_work.work.handler = tmp108_trigger_handle_one_shot;
+
+	/* save this driver instance for passing to other fuunctions */
+	drv_data->tmp108_dev = dev;
+
+#ifdef  CONFIG_TMP108_ALERT_INTERRUPTS
+	setup_interrupts(dev);
+#endif
+
+	return 0;
+}
+
+static struct tmp108_data tmp108_prv_data;
+
+DEVICE_DT_INST_DEFINE(0,
+		      tmp108_init,
+		      NULL,
+		      &tmp108_prv_data,
+		      NULL,
+		      POST_KERNEL,
+		      CONFIG_SENSOR_INIT_PRIORITY,
+		      &tmp108_driver_api);
```

Comment:
Please update to support multiple instances by using `DT_INST_FOREACH_STATUS_OKAY`. See https://docs.zephyrproject.org/latest/guides/dts/howtos.html#write-device-drivers-using-devicetree-apis for details.